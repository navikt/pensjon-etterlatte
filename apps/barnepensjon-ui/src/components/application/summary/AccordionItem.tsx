import { EditFilled } from '@navikt/ds-icons'
import { Accordion } from '@navikt/ds-react'
import { useState } from 'react'
import { Link } from 'react-router-dom'

export const AccordionItem = ({ title, children, defaultOpen = false, path, pathText }: any) => {
    const [open, setOpen] = useState(defaultOpen)

    return (
        <Accordion.Item open={open}>
            <Accordion.Header
                onClick={(e) => {
                    e.preventDefault()
                    setOpen(!open)
                }}
                aria-expanded={open}
            >
                {title}
            </Accordion.Header>
            <Accordion.Content>
                {children}
                <Link to={path}>
                    <EditFilled />
                    <span style={{paddingLeft: '0.25rem'}}>{pathText}</span>
                </Link>
            </Accordion.Content>
        </Accordion.Item>
    )
}
