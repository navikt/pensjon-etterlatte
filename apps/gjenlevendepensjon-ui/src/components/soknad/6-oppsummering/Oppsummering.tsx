import { Alert, BodyLong, Button, Heading, Link, Loader } from '@navikt/ds-react'
import { isEmpty } from "lodash";
import { SkjemaGruppe } from "../../felles/SkjemaGruppe";
import React, { memo, useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { useNavigate } from "react-router-dom";
import { sendSoeknad } from "../../../api/api";
import { SoeknadRequest, SoeknadType } from "../../../api/dto/InnsendtSoeknad";
import { mapTilBarnepensjonSoeknadListe, mapTilGjenlevendepensjonSoeknad } from "../../../api/mapper/soeknadMapper";
import { useBrukerContext } from "../../../context/bruker/BrukerContext";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import SoknadSteg from "../../../typer/SoknadSteg";
import { LogEvents, useAmplitude } from "../../../utils/amplitude";
import SoeknadMapper from "../../../utils/SoeknadMapper";
import Navigasjon from "../../felles/Navigasjon";
import OppsummeringInnhold from "./OppsummeringInnhold";
import { ActionTypes } from "../../../context/soknad/soknad";
import {NavigasjonsRad, SpoersmaalModal} from "../../felles/StyledComponents";
import { SkjemaElement } from "../../felles/SkjemaElement";

const Oppsummering: SoknadSteg = memo(({ forrige }) => {
    const navigate = useNavigate();
    const [soeknadOppsummering, setOppsummering] = useState<any>([]);
    const { t } = useTranslation();

    const { state: soeknad, dispatch } = useSoknadContext();
    const { state: bruker } = useBrukerContext();
    const { logEvent } = useAmplitude();

    const [senderSoeknad, setSenderSoeknad] = useState(false);
    const [error, setError] = useState(false);
    const [isOpen, setIsOpen] = useState(false);

    const mapper = new SoeknadMapper(t);

    useEffect(() => {
        (async () => {
            if (isEmpty(soeknad) || isEmpty(bruker)) {
                setOppsummering([]);
            } else {
                const soeknadOppsummering = mapper.lagOppsummering(soeknad, bruker);
                setOppsummering(soeknadOppsummering);
            }
        })();
    }, [soeknad, bruker]);

    const send = () => {
        setSenderSoeknad(true);
        setError(false);

        const gjenlevendepensjon = mapTilGjenlevendepensjonSoeknad(t, soeknad, bruker);
        const barnepensjonSoeknader = mapTilBarnepensjonSoeknadListe(t, soeknad, bruker);

        const soeknadBody: SoeknadRequest = {
            soeknader: [gjenlevendepensjon, ...barnepensjonSoeknader],
        };

        sendSoeknad(soeknadBody)
            .then(() => {
                logEvent(LogEvents.SEND_SOKNAD, { type: SoeknadType.GJENLEVENDEPENSJON });

                barnepensjonSoeknader.forEach(() => {
                    logEvent(LogEvents.SEND_SOKNAD, { type: SoeknadType.BARNEPENSJON });
                });

                dispatch({ type: ActionTypes.TILBAKESTILL });
                navigate(`/skjema/sendt`);
            })
            .catch((error) => {
                setIsOpen(false)
                console.log(error);
                setSenderSoeknad(false);
                setError(true);
            });
    };

    return (
        <>
            <SkjemaElement>
                <Heading size={"medium"} className={"center"}>
                    {t("oppsummering.tittel")}
                </Heading>
            </SkjemaElement>

            <SkjemaElement>
                <BodyLong>{t("oppsummering.beskrivelse")}</BodyLong>
            </SkjemaElement>

            {!isEmpty(soeknadOppsummering) && (
                <OppsummeringInnhold soeknadOppsummering={soeknadOppsummering} senderSoeknad={senderSoeknad} />
            )}

            <br />

            {error && (
                <SkjemaGruppe>
                    <Alert variant={"error"}>
                        {t("oppsummering.feilVedSending")}
                        <Link href={t("oppsummering.feilVedSending.href")}>
                            {t("oppsummering.feilVedSending.tittel")}
                        </Link>
                    </Alert>
                </SkjemaGruppe>
            )}

            <Navigasjon forrige={{ onClick: forrige }} send={{ onClick: () => setIsOpen(true) }} disabled={senderSoeknad} />

            <SpoersmaalModal open={isOpen}
                    onClose={() => {
                        if (!senderSoeknad) setIsOpen(false)
                    }}
                shouldCloseOnOverlayClick={false}
                 data-testid={"spoersmaal-modal"}
                >


                <SkjemaElement>
                    <Heading size={'medium'}>{t(senderSoeknad ? 'oppsummering.senderSoeknad.tittel' : 'oppsummering.sendSoeknad.tittel')}</Heading>
                </SkjemaElement>

                <SkjemaElement>
                    {senderSoeknad ? (
                            <Loader size={'xlarge'} />
                    ) : (
                            <BodyLong size={'small'}>{t('oppsummering.sendSoeknad.innhold')}</BodyLong>
                    )}
                </SkjemaElement>
                {!senderSoeknad && (
                        <NavigasjonsRad>
                            <Button
                                    id={'avbryt-ja-btn'}
                                    variant={'secondary'}
                                    type={'button'}
                                    onClick={() => setIsOpen(false)}
                                    style={{ margin: '10px' }}
                            >
                                {t('knapp.nei')}
                            </Button>
                            <Button
                                    id={'avbryt-nei-btn'}
                                    variant={'primary'}
                                    type={'button'}
                                    onClick={send}
                                    style={{ margin: '10px' }}
                            >
                                {t('knapp.ja')}
                            </Button>
                        </NavigasjonsRad>
                )}
            </SpoersmaalModal>
        </>
    );
});

export default Oppsummering;
