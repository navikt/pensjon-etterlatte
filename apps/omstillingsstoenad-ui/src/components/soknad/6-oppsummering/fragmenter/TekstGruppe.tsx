import { BodyShort, Label } from '@navikt/ds-react'

const stringify = (innhold?: any) => {
    if (!innhold) return ''
    else if (typeof innhold !== 'string') return innhold.toString()
    else return innhold
}

const TekstGruppe = ({ tittel, innhold, id }: { tittel: string; innhold?: any; id?: string }) => {
    return (
        <div className={'tekstgruppe'}>
            <Label>{tittel}</Label>
            <BodyShort id={id}>{stringify(innhold)}</BodyShort>
        </div>
    )
}

export default TekstGruppe
