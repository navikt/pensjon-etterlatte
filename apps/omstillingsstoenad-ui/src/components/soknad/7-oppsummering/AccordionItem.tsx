import { Accordion } from '@navikt/ds-react'
import { useState } from 'react'

export const AccordionItem = ({ tittel, children, defaultOpen = false }: any) => {
    const [open, setOpen] = useState(defaultOpen)

    return (
        <Accordion.Item open={open}>
            <Accordion.Header
                onClick={(e) => {
                    e.preventDefault()
                    setOpen(!open)
                }}
                aria-expanded={open}
            >
                {tittel}
            </Accordion.Header>
            <Accordion.Content>{children}</Accordion.Content>
        </Accordion.Item>
    )
}
