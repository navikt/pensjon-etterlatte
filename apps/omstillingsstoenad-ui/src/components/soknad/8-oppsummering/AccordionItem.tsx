import { Accordion } from '@navikt/ds-react'
import { useState } from 'react'
import { EditFilled } from '@navikt/ds-icons'
import { Link } from 'react-router-dom'
import { useTranslation } from 'react-i18next'

export const AccordionItem = ({ tittel, children, defaultOpen = false, path, pathText, senderSoeknad }: any) => {
    const { t } = useTranslation()
    const [open, setOpen] = useState(defaultOpen)

    return (
        <Accordion.Item open={open}>
            <Accordion.Header
                onClick={(e) => {
                    e.preventDefault()
                    setOpen(!open)
                }}
                aria-expanded={open}
            >
                {tittel}
            </Accordion.Header>
            <Accordion.Content>
                {children}
                <Link to={path} className={senderSoeknad ? 'disabled' : ''}>
                    <EditFilled />
                    <span style={{ paddingLeft: '0.25rem' }}>{t(`endreSvarOppsummering.${pathText}`)}</span>
                </Link>
            </Accordion.Content>
        </Accordion.Item>
    )
}
