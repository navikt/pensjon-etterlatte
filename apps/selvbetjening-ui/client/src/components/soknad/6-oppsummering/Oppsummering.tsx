import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { useHistory } from "react-router-dom";
import React, { memo, useEffect, useState } from "react";
import { SkjemaGruppe } from "nav-frontend-skjema";
import SoknadSteg from "../../../typer/SoknadSteg";
import { Alert, BodyLong, Heading } from "@navikt/ds-react";
import Navigasjon from "../../felles/Navigasjon";
import { useTranslation } from "react-i18next";
import { useBrukerContext } from "../../../context/bruker/BrukerContext";
import { hentOppsummering, sendSoeknad } from "../../../api/api";
import OppsummeringInnhold from "./OppsummeringInnhold";
import { isEmpty } from "lodash";
import { LogEvents, useAmplitude } from "../../../utils/amplitude";
import {
    mapTilBarnepensjonSoeknadListe,
    mapTilGjenlevendepensjonSoeknad
} from "../../../api/mapper/soeknadMapper";
import { SoeknadRequest, SoeknadType } from "../../../api/dto/InnsendtSoeknad";

const Oppsummering: SoknadSteg = memo(({ forrige }) => {
    const history = useHistory();
    const [soeknadOppsummering, setOppsummering] = useState<any>([]);
    const { t, i18n } = useTranslation();

    const { state: soeknad } = useSoknadContext();
    const { state: bruker } = useBrukerContext();
    const { logEvent } = useAmplitude();

    const [senderSoeknad, setSenderSoeknad] = useState(false);
    const [error, setError] = useState(false);

    useEffect(() => {
        (async () => {
            if (isEmpty(soeknad) || isEmpty(bruker)) setOppsummering([]);
            const soeknadOppsummering: any = await hentOppsummering({soeknad, bruker, locale: i18n.language});
            setOppsummering(soeknadOppsummering);
        })()
    }, [soeknad, bruker])

    const send = () => {
        setSenderSoeknad(true);
        setError(false);

        const gjenlevendepensjon = mapTilGjenlevendepensjonSoeknad(t, soeknad, bruker);
        const barnepensjonSoeknader = mapTilBarnepensjonSoeknadListe(t, soeknad, bruker);

        const soeknadBody: SoeknadRequest = {
            soeknader: [gjenlevendepensjon, ...barnepensjonSoeknader]
        };

        sendSoeknad(soeknadBody)
            .then(() => {
                logEvent(LogEvents.SEND_SOKNAD, { type: SoeknadType.GJENLEVENDEPENSJON });

                barnepensjonSoeknader.forEach(() => {
                    logEvent(LogEvents.SEND_SOKNAD, { type: SoeknadType.BARNEPENSJON });
                });

                history.push(`/skjema/sendt`);
            })
            .catch((error) => {
                console.log(error);
                setSenderSoeknad(false);
                setError(true);
            });
    };

    return (
        <>
            <SkjemaGruppe>
                <Heading size={"medium"} className={"center"}>
                    {t("oppsummering.tittel")}
                </Heading>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <BodyLong>{t("oppsummering.beskrivelse")}</BodyLong>
            </SkjemaGruppe>

            <OppsummeringInnhold
                soeknadOppsummering={soeknadOppsummering}
                senderSoeknad={senderSoeknad}
            />

            <br />

            {error && (
                <SkjemaGruppe>
                    <Alert variant={"error"}>{t("oppsummering.feilVedSending")}</Alert>
                </SkjemaGruppe>
            )}

            <Navigasjon forrige={{ onClick: forrige }} send={{ onClick: send }} disabled={senderSoeknad} />
        </>
    );
});

export default Oppsummering;
