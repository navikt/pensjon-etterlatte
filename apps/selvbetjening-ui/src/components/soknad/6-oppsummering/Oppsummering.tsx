import { Alert, BodyLong, Heading, Link } from "@navikt/ds-react";
import { isEmpty } from "lodash";
import { SkjemaGruppe } from "nav-frontend-skjema";
import React, { memo, useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { useHistory } from "react-router-dom";
import { sendSoeknad } from "../../../api/api";
import { SoeknadRequest, SoeknadType } from "../../../api/dto/InnsendtSoeknad";
import { mapTilBarnepensjonSoeknadListe, mapTilGjenlevendepensjonSoeknad } from "../../../api/mapper/soeknadMapper";
import { useBrukerContext } from "../../../context/bruker/BrukerContext";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import SoknadSteg from "../../../typer/SoknadSteg";
import { LogEvents, useAmplitude } from "../../../utils/amplitude";
import SoeknadMapper from "../../../utils/SoeknadMapper";
import Navigasjon from "../../felles/Navigasjon";
import OppsummeringInnhold from "./OppsummeringInnhold";

const Oppsummering: SoknadSteg = memo(({ forrige }) => {
    const history = useHistory();
    const [soeknadOppsummering, setOppsummering] = useState<any>([]);
    const { t } = useTranslation();

    const { state: soeknad } = useSoknadContext();
    const { state: bruker } = useBrukerContext();
    const { logEvent } = useAmplitude();

    const [senderSoeknad, setSenderSoeknad] = useState(false);
    const [error, setError] = useState(false);

    const mapper = new SoeknadMapper(t);

    useEffect(() => {
        (async () => {
            if (isEmpty(soeknad) || isEmpty(bruker)) {
                setOppsummering([]);
            } else {
                const soeknadOppsummering = mapper.lagOppsummering(soeknad, bruker);
                setOppsummering(soeknadOppsummering);
            }
        })();
    }, [soeknad, bruker]);

    const send = () => {
        setSenderSoeknad(true);
        setError(false);

        const gjenlevendepensjon = mapTilGjenlevendepensjonSoeknad(t, soeknad, bruker);
        const barnepensjonSoeknader = mapTilBarnepensjonSoeknadListe(t, soeknad, bruker);

        const soeknadBody: SoeknadRequest = {
            soeknader: [gjenlevendepensjon, ...barnepensjonSoeknader],
        };

        sendSoeknad(soeknadBody)
            .then(() => {
                logEvent(LogEvents.SEND_SOKNAD, { type: SoeknadType.GJENLEVENDEPENSJON });

                barnepensjonSoeknader.forEach(() => {
                    logEvent(LogEvents.SEND_SOKNAD, { type: SoeknadType.BARNEPENSJON });
                });

                history.push(`/skjema/sendt`);
            })
            .catch((error) => {
                console.log(error);
                setSenderSoeknad(false);
                setError(true);
            });
    };

    return (
        <>
            <SkjemaGruppe>
                <Heading size={"medium"} className={"center"}>
                    {t("oppsummering.tittel")}
                </Heading>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <BodyLong>{t("oppsummering.beskrivelse")}</BodyLong>
            </SkjemaGruppe>

            {!isEmpty(soeknadOppsummering) && (
                <OppsummeringInnhold soeknadOppsummering={soeknadOppsummering} senderSoeknad={senderSoeknad} />
            )}

            <br />

            {error && (
                <SkjemaGruppe>
                    <Alert variant={"error"}>
                        {t("oppsummering.feilVedSending")}
                        <Link href={t("oppsummering.feilVedSending.href")}>
                            {t("oppsummering.feilVedSending.tittel")}
                        </Link>
                    </Alert>
                </SkjemaGruppe>
            )}

            <Navigasjon forrige={{ onClick: forrige }} send={{ onClick: send }} disabled={senderSoeknad} />
        </>
    );
});

export default Oppsummering;
