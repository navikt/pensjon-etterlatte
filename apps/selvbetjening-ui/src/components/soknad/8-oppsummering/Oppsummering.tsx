import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { sendSoeknad } from "../../../api";
import { useHistory } from "react-router-dom";
import { Ingress, Normaltekst, Systemtittel } from "nav-frontend-typografi";
import React, { useState } from "react";
import { Fareknapp, Flatknapp, Hovedknapp, Knapp } from "nav-frontend-knapper";
import { SkjemaGruppe } from "nav-frontend-skjema";
import OppsummeringAvdoed from "./fragmenter/OppsummeringAvdoed";
import SoknadSteg from "../../../typer/SoknadSteg";
import { useTranslation } from "react-i18next";
import OppsummeringBarn from "./fragmenter/OppsummeringBarn";
import { default as Modal } from "nav-frontend-modal";
import { ActionTypes } from "../../../context/soknad/soknad";
import NavFrontendSpinner from "nav-frontend-spinner";
import AlertStripe from "nav-frontend-alertstriper";
import classNames from "classnames";
import OppsummeringOmDeg from "./fragmenter/OppsummeringOmDeg";

const Oppsummering: SoknadSteg = ({ forrige }) => {
    const history = useHistory();
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();

    const {
        omDeg,
        omDenAvdoede,
        opplysningerOmBarn,
    } = state;

    const [isOpen, setIsOpen] = useState(false);

    const [senderSoeknad, setSenderSoeknad] = useState(false);
    const [error, setError] = useState(false);

    const send = () => {
        setSenderSoeknad(true);

        sendSoeknad(state)
            .then((soknadId) => {
                history.push(`/soknad/sendt/${soknadId}`);
            })
            .catch((error) => {
                console.log(error)
                setSenderSoeknad(false);
                setError(true)
            });
    };

    const avbrytSoeknad = () => {
        dispatch({ type: ActionTypes.TILBAKESTILL })

        window.location.href = "https://www.nav.no"
    }

    return (
        <>
            <SkjemaGruppe>
                <Systemtittel className={"center"}>Oppsummering</Systemtittel>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <Normaltekst>Les gjennom oppsummeringen av din søknad før du sender.</Normaltekst>
                <Normaltekst>Hvis du trenger å gjøre endringer, kan du gå tilbake og gjøre det. </Normaltekst>
            </SkjemaGruppe>

            <OppsummeringOmDeg state={omDeg!!} />

            <OppsummeringAvdoed state={omDenAvdoede!!} />

            <OppsummeringBarn state={opplysningerOmBarn} />

            <br />

            <Modal
                isOpen={isOpen}
                onRequestClose={() => setIsOpen(false)}
                closeButton={true}
                contentLabel={"aasdfsdf"}
            >
                <SkjemaGruppe>
                    <Ingress>
                        Er du helt sikker på at du vil avbryte søknaden?
                    </Ingress>
                </SkjemaGruppe>

                <SkjemaGruppe>
                    <Hovedknapp htmlType={"button"} onClick={() => setIsOpen(false)}>
                        Nei, jeg vil fortsette søknaden
                    </Hovedknapp>
                </SkjemaGruppe>

                <SkjemaGruppe>
                    <Fareknapp htmlType={"button"} onClick={avbrytSoeknad}>
                        Ja, avbryt søknaden
                    </Fareknapp>
                </SkjemaGruppe>
            </Modal>

            {error && (
                <SkjemaGruppe>
                    <AlertStripe type={"feil"}>
                        En feil oppsto ved sending. Vent litt og prøv på nytt. Dersom feilen vedvarer kan du kontakte kundeservice.
                    </AlertStripe>
                </SkjemaGruppe>
            )}

            <SkjemaGruppe className={classNames("navigasjon-rad", senderSoeknad && "disabled")} >
                <Knapp htmlType={"button"} onClick={forrige}>
                    {t("knapp.tilbake")}
                </Knapp>

                <Hovedknapp htmlType={"button"} onClick={send}>
                    {t("knapp.sendSoeknad")} {senderSoeknad && (
                        <NavFrontendSpinner />
                    )}
                </Hovedknapp>
            </SkjemaGruppe>

            <SkjemaGruppe className={classNames("navigasjon-rad", senderSoeknad && "disabled")}>
                <Flatknapp htmlType={"button"} onClick={() => setIsOpen(true)}>
                    Avbryt
                </Flatknapp>
            </SkjemaGruppe>
        </>
    );
};

export default Oppsummering;
