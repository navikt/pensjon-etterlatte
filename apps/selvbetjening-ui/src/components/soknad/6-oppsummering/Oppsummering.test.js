import { render, act, fireEvent } from "@testing-library/react";
import * as JSutils from "nav-frontend-js-utils";
import { AccordionItem } from "./AccordionItem";
import TekstGruppe from "./fragmenter/TekstGruppe";
import Oppsummering from "./Oppsummering";

jest.mock("react-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: jest.fn(() => ({ pathname: "/skjema/steg/oppsummering" })),
}));

JSutils.guid = jest.fn(() => "123");
describe("Oppsummering", () => {
    it("Snapshot", () => {
        const { container } = render(<Oppsummering />);
        expect(container).toMatchSnapshot();
    });
});

describe("Test accordionItem", () => {
    it("skal endre aria-expanded", () => {
        const { container, getByText } = render(
            <AccordionItem tittel="Testtittel" defaultOpen={false}>
                Innhold
            </AccordionItem>
        );
        expect(container.querySelectorAll("[aria-expanded]")[0].getAttribute("aria-expanded")).toBe("false");
        fireEvent.click(getByText("Testtittel"));
        expect(container.querySelectorAll("[aria-expanded]")[0].getAttribute("aria-expanded")).toBe("true");
    });
    it("skal rendre innholdet synlig", () => {
        const { getByText } = render(
            <AccordionItem tittel="Testtittel" defaultOpen={true}>
                Innhold
            </AccordionItem>
        );
        expect(getByText("Innhold")).toBeDefined()
    });
});

describe("Tekstgruppe", () => {
    it("Skal rendre testittel og testcontent", () => {
        const { getByText } = render(<TekstGruppe tittel="Testtittel" innhold={"Testcontent"} />);
        expect(getByText("Testtittel")).toBeDefined();
        expect(getByText("Testcontent")).toBeDefined();
    })
})