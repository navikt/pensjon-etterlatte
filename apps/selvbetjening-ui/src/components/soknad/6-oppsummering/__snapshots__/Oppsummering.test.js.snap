// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Oppsummering Snapshot 1`] = `
<div>
  <fieldset
    aria-invalid="false"
    class="skjemagruppe"
  >
    <h1
      class="center navds-heading navds-heading--medium"
    >
      oppsummering.tittel
    </h1>
    <div
      aria-live="polite"
    />
  </fieldset>
  <fieldset
    aria-invalid="false"
    class="skjemagruppe"
  >
    <p
      class="navds-body-long"
    >
      oppsummering.beskrivelse
    </p>
    <div
      aria-live="polite"
    />
  </fieldset>
  <br />
  <fieldset
    aria-invalid="false"
    class="skjemagruppe navigasjon-footer"
  >
    <div
      class="navigasjon-rad"
    >
      <button
        class="navds-button navds-button--primary navds-button--medium"
        type="button"
      >
        <span
          class="navds-button__inner navds-body-short"
        >
          knapp.tilbake
        </span>
      </button>
      <button
        class="navds-button navds-button--primary navds-button--medium"
        type="button"
      >
        <span
          class="navds-button__inner navds-body-short"
        >
          knapp.sendSoeknad
           
        </span>
      </button>
    </div>
    <div
      aria-live="polite"
    />
  </fieldset>
  <fieldset
    aria-invalid="false"
    class="skjemagruppe navigasjon-rad"
  >
    <button
      class="navds-button navds-button--secondary navds-button--medium"
      id="avbryt-btn"
      type="button"
    >
      <span
        class="navds-button__inner navds-body-short"
      >
        knapp.avbryt
      </span>
    </button>
    <div
      aria-live="polite"
    />
  </fieldset>
</div>
`;
